### PR TITLE
Initial refactor for domains and providers

### LINKED ISSUE #332: Refactor domain and resources
The current validation structure has a somewhat confusing use of `domain` and `resources`, where they appear separate in the definition, but are coupled in practice - e.g.,
```yaml
target:
  provider: opa
  domain: kubernetes
  payload:
    resources:
      - {list of resources}
```

This flows into the code organization - e.g., getting the resources is implemented redundantly inside the provider modules.

The structure of both the validation yaml and implementation in code should probably be refactored to reduce redundant usage and enhance clarity.

### PR DESCRIPTION
**Pull request by meganwolf0**
Some initial stabs at refactoring the code to support domains and providers in a (hopefully) more extensible and modular way

### CODE REVIEW

File: src/pkg/common/types.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package common
+
+import (
+	"github.com/defenseunicorns/lula/src/pkg/domains/api"
+	kube "github.com/defenseunicorns/lula/src/pkg/domains/kubernetes"
+	"github.com/defenseunicorns/lula/src/pkg/providers/kyverno"
+	"github.com/defenseunicorns/lula/src/pkg/providers/opa"
+)
+
+// YAML data structures for ingesting validation data
```
Comment (brandtkeller) [inline src/pkg/common/types.go]: Does any of this really lock us in to yaml? wondering what would need to be modified in order to support JSON and if that would change the naming here?

File: src/types/types.go [hunk 2]
```diff
@@ -1,100 +1,46 @@
 package types
 
-import (
-	"errors"
-	kjson "github.com/kyverno/kyverno-json/pkg/apis/policy/v1alpha1"
+type LulaValidationType string
+
+const (
+	LulaValidationTypeNormal  LulaValidationType = "Lula Validation"
```
Comment (brandtkeller) [inline src/types/types.go:6]: let's discuss validation "types" more. Curious if that is still within the boundary of validations or not.

File: src/pkg/common/types.go (hunk 1 above)
Comment (meganwolf0) [reply src/pkg/common/types.go]: No I think you're right, this probably shouldn't be called "yaml" - I was trying to figure out how to delineate between the pieces that would be like I/O vs the lula-internal structures (which I tried to bucket into types/types.go) but happy to rename to what makes sense there

File: src/pkg/providers/kyverno/types.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+func (k KyvernoProvider) Evaluate(resources types.DomainResources) (types.Result, error) {
+	results, err := GetValidatedAssets(k.Context, k.Spec.Kyverno, resources, k.Spec.Output)
+	if err != nil {
+		return types.Result{}, err
+	}
+	return results, nil
+}
+
+type KyvernoSpec struct {
+	Kyverno *kjson.ValidatingPolicy `json:"kyverno" yaml:"kyverno"`
```
Comment (brandtkeller) [inline src/pkg/providers/kyverno/types.go]: No change required - thinking about if `kyverno` is the most intuitive option here. 

`opa-spec` has `rego` 

Should `kyverno-spec` have `policy` ?

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Needs `docs/` files updated.

**Review by mike-winberry — COMMENTED** [review]
Looks good, just have one item for discussion on #354

**Review by mike-winberry — APPROVED** [review]

**Review by brandtkeller — APPROVED** [review]

**Pull request merged**